Extension/src/LanguageServer/client.ts

```diff
@@ -620,6 +628,25 @@ class DefaultClient implements Client {
         this.model.tagParserStatus.Value = notificationBody.status;
     }
 
+    private updateInactiveRegions(params: InactiveRegionParams): void {
+        let renderOptions: vscode.DecorationRenderOptions = {
+            light: { color: "rgba(125,125,125,1.0)" },
+            dark: { color: "rgba(155,155,155,1.0)" }
+        };
+
+        // Recycle the active text decorations when we receive a new set of inactive regions
+        if (this.inactiveRegionsDecoration !== undefined) {
+            this.inactiveRegionsDecoration.dispose();
```

Comment:
Thinking about this a bit more...  Since the inactiveRegionsDecorarion is linked to the Client, this will end up accidentally deleting highlights from files when you have more than one file open in the editor.  You're going to want to have one of these for each uri that you encounter.

And then you will want to add disposal code to DefaultClient.dispose